File: Cpp/9_Palindrome_Number.cpp
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;

        int rev = 0;
        int n = x;

        while (n != 0) {
            rev = rev * 10 + n % 10;
            n /= 10;
        }

        return (rev == x);
    }
}

File: Python/9_Palindrome_Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False
        r, n = 0, x
        while x > 0:
            digit = x % 10
            r = r * 10 + digit
            x //= 10
        return n == r
        
